### PR TITLE
fix(DCOS_OSS-1272): retain port value if requirePorts is set to false

### PR DESCRIPTION
**Pull request by bstavroulakis**
I removed the this.appState.portsAutoAssign check.

To reproduce:

1) Copy the following JSON in a new Service JSON Form.
2) Run the service.
3) Edit the service, the port should remain 10001 and not reset to 0 even though requirePorts is false.

```
{
    "id": "/chronos",
    "backoffFactor": 1.15,
    "backoffSeconds": 1,
    "container": {
      "type": "DOCKER",
      "docker": {
        "forcePullImage": false,
        "image": "nginx"
      }
    },
    "cpus": 1,
    "disk": 0,
    "instances": 1,
    "maxLaunchDelaySeconds": 3600,
    "mem": 1024,
    "gpus": 0,
    "networks": [
      {
        "mode": "host"
      }
    ],
    "portDefinitions": [
      {
        "port": 10001,
        "protocol": "tcp"
      }
    ],
    "requirePorts": false
}
```

Closes DCOS_OSS-1272

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by nLight** [comment]
Hey Bill, I'm sorry that slipped my mind, I think we should also change the review screen. So that when there's a port value along with `requirePorts: false` we should say `Host Port = value`

screen shot 2017-07-11 at 11 41 32

So we either want to stick to 0, or say that it's going to be assigned automatically in this particular case.

**Comment by bstavroulakis** [comment]
@nLight The review screen has been addressed as well.

**Comment by nLight** [comment]
Hey @gkleiman @timcharper please tell us what is correct in case of having `port` set to something and `requirePorts` set to `false`. What do we show as a Host Port? Is it `0` or something like `Auto Assigned` or `N/A`? https://github.com/dcos/dcos-ui/pull/2307#issuecomment-314393919

**Comment by gkleiman** [comment]
@nLight: in my opinion something like `Auto Assigned` would be more user-friendly, but I'll let Tim chime in =).

**Comment by bstavroulakis** [comment]
@nLight @gkleiman @timcharper: When the `requirePorts` is set to `false` in the configuration now instead of 0 we show the "Auto Assigned" message.

File: plugins/services/src/js/service-configuration/ServiceNetworkingConfigSection.js [hunk 1]
```diff
@@ -114,7 +114,9 @@ class ServiceNetworkingConfigSection extends ServiceConfigBaseSectionDisplay {
                 prop: keys.port,
                 className: getColumnClassNameFn(),
                 render(prop, row) {
-                  return getDisplayValue(row[prop]);
+                  return appDefinition.requirePorts
+                    ? getDisplayValue(row[prop])
+                    : <em>Auto Assigned</em>;
```
Comment (weblancaster) [inline plugins/services/src/js/service-configuration/ServiceNetworkingConfigSection.js]: @bstavroulakis can you confirm `em` (emphasis) was approved by design please? other than that works perfectly.

Comment (bstavroulakis) [inline plugins/services/src/js/service-configuration/ServiceNetworkingConfigSection.js]: @ashenden I've sent you a message with an image snapshot of this.

screen shot 2017-07-17 at 10 17 02 am

**Comment by bstavroulakis** [comment]
@nLight @weblancaster All changes done, tested and approved by the design team.

**Pull request merged**